src/libsystemd/sd-bus/bus-socket.c

```diff
@@ -1406,35 +1406,41 @@ int bus_socket_read_message(sd_bus *bus) {
 
         if (handle_cmsg) {
                 struct cmsghdr *cmsg;
+                bool failed = false;
 
                 CMSG_FOREACH(cmsg, &mh)
                         if (cmsg->cmsg_level == SOL_SOCKET &&
                             cmsg->cmsg_type == SCM_RIGHTS) {
-                                int n, *f, i;
-
-                                n = (cmsg->cmsg_len - CMSG_LEN(0)) / sizeof(int);
-
                                 if (!bus->can_fds) {
                                         /* Whut? We received fds but this
                                          * isn't actually enabled? Close them,
                                          * and fail */
 
-                                        close_many(CMSG_TYPED_DATA(cmsg, int), n);
+                                        cmsg_close_all(&mh);
```

Comment:
btw, cmsg_close_all() is probably what we should always use in cases like this, since it takes care of closing both SCM_RIGHTS and SCM_PIDFD.